pynxtools/dataconverter/readers/em/geometry/handed_cartesian.py

```diff
@@ -0,0 +1,749 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Definitions for right- or left-handed Cartesian coordinate systems."""
+
+# pylint: disable=no-member
+
+# a coordinate system can be defined using an instance of a custom schema
+# with a set of sets of RadioEditButtons, each set can be used to define the
+# unique orientation of a right-handed Cartesian coordinate system's base vectors when
+# viewed by an outer observer looking towards the origin of the coordinate system
+# triplet vector, specifically we inspect the line of sight and a compass
+# the compass defines four directions, north aka up, east aka right, south aka down,
+# and west aka left. The plane of sight defines two additional directions in aka
+# into the plane and out aka out-of-the-plane
+# there are different strategies what users find convenient to report to specify
+# the directions into which the base vectors are assumed pointing
+
+# in general only two perpendicular directions have to be specified as the third
+# follows implicitly i.e. there is no need to overconstrain
+# the term "undefined" is the default to use unless one is sure about each statement
+
+# the is_cs_rh_unambiguous dictionary is a lookup table whereby the input of the user
+# can be evaluated to check if the radioeditbutton choices result in a fully constraint
+# and thus unique definition. All other choices are reported as a not fully constraint
+# i.e. eventually problematic coordinate system
+
+REFERENCE_FRAMES = ["undefined", "right_handed_cartesian", "left_handed_cartesian"]
+
+AXIS_DIRECTIONS = ["undefined", "north", "east", "south", "west", "in", "out"]
+
+# is a right-handed (rh) Cartesian coordinate system (cs) sufficiently constrained, when
+# at least two base vector directions are chosen, the order is x, y, z
+is_cs_rh_unambiguous = {"undefined_undefined_undefined": False,
```

Comment:
You're using dicts almost as functions. This is quite an irregular syntax. Consider using lamba functions or the sorts for this if you wanna save space.

Reply:
This is a look-up table